core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ApplicationSummaryInfo.scala

```diff
@@ -136,6 +140,27 @@ class SingleAppSummaryInfoProvider(val app: ApplicationSummaryInfo)
     }
   }
 
+  override def getSpillMetricsForStage(stage: String): Seq[(Long, Long)] = {
+    // Get ids for input specified stage
+    val stageIds = app.sqlStageInfo.collect ({
+      case row if (row.nodeNames.exists(name => name.toLowerCase.contains(stage))) =>
+        row.stageId.toString()
+    })
+
+    // Select those in the specified stage and retrieve their spilled metrics
+    app.jsMetAgg.collect({
+      case row if (row.id.contains("stage") && stageIds.contains(row.id.split("_")(1))) =>
```

Comment:
you can modify stageIds to return "stage_STAGEID" then you can do a direct match

```scala
    val stageIds = app.sqlStageInfo.collect ({
      case row if (row.nodeNames.exists(name => name.toLowerCase.contains(stage))) =>
        s"stage_${row.stageId.toString()}"
    }).toSet
    // Select those in the specified stage and retrieve their spilled metrics
    app.jsMetAgg.collect({
      case row if stageIds.contains(row.id) =>
        (row.id.split("_")(1).toLong, row.diskBytesSpilledSum + row.memoryBytesSpilledSum)
    })
```

Reply:
I do not see this change in the latest update.

Reply:
I do not see this change in the latest update.

Reply:
@parthosa I changed the logic so the code is different now. Thanks.